specs/_features/eip7668/beacon_chain.md

```diff
@@ -0,0 +1,88 @@
+Limit churn -- The Beacon Chain
+
+## Table of contents
+
+<!-- TOC -->
+<!-- START doctoc generated TOC please keep comment here to allow auto update -->
+<!-- DON'T EDIT THIS SECTION, INSTEAD RE-RUN doctoc TO UPDATE -->
+
+- [Introduction](#introduction)
+- [Configuration](#configuration)
+  - [Validator cycle](#validator-cycle)
+- [Helper functions](#helper-functions)
+  - [Beacon state accessors](#beacon-state-accessors)
+    - [New `get_validator_inbound_churn_limit`](#new-get_validator_inbound_churn_limit)
+- [Beacon chain state transition function](#beacon-chain-state-transition-function)
+  - [Epoch processing](#epoch-processing)
+    - [Registry updates](#registry-updates)
+
+<!-- END doctoc generated TOC please keep comment here to allow auto update -->
+<!-- /TOC -->
+
+## Introduction
+
+This is the beacon chain specification to limit the max inbound churn value, motivated to limit the validator active set growth rate.
+
+*Note:* This specification is built upon [Capella](../../capella/beacon_chain.md) and is under active development.
+
+## Configuration
+
+### Validator cycle
+
+| Name | Value |
+| - | - |
+| `MAX_PER_EPOCH_INBOUND_CHURN_LIMIT` | `uint64(12)` (= 12) |
+
+## Helper functions
+
+### Beacon state accessors
+
+#### New `get_validator_inbound_churn_limit`
+
+```python
+def get_validator_inbound_churn_limit(state: BeaconState) -> uint64:
+    """
+    Return the validator inbound churn limit for the current epoch.
+    """
+    active_validator_indices = get_active_validator_indices(state, get_current_epoch(state))
+    return min(
+        MAX_PER_EPOCH_INBOUND_CHURN_LIMIT,
+        max(
```

Comment:
This could call `get_validator_churn_limit()` as the second param of the `min()` function to avoid duplicating the `max` logic in two places.

I also suggest naming this `get_validator_activation_churn_limit()` to map to the name we use for this entry

Also, we might consider adding a function `get_validator_exit_churn_limit()` and replacing `get_validator_churn_limit()` with it to have the naming parallelism. don't feel strongly about this.

Reply:
+1 on calling `get_validator_churn_limit`

Reply:
> Also, we might consider adding a function get_validator_exit_churn_limit() and replacing get_validator_churn_limit() with it to have the naming parallelism. don't feel strongly about this.

Leveraging the psychology of minimal diff